### PR TITLE
Themes JS does not load in preview mode issue  fixed

### LINKED ISSUE #186: Themes JS does not load in preview mode

### PR DESCRIPTION
**Pull request by swanandpachode94**
Closes #186

### CODE REVIEW

**Review by Gaganbird — APPROVED** [review]

File: Portals/_default/Skins/Vanjaro/Base.ascx.cs [hunk 1]
```diff
@@ -124,6 +124,10 @@ protected void Page_Load(object sender, EventArgs e)
         {
             JavaScript.RequestRegistration(CommonJs.jQuery, null, SpecificVersion.Latest);
 
+            string ThemeJS = "~/Portals/_default/vThemes/" + Core.Managers.ThemeManager.CurrentTheme.Name + "/theme.js";
+            if (File.Exists(System.Web.Hosting.HostingEnvironment.MapPath(ThemeJS)))
```
Comment (meetmandeep) [inline Portals/_default/Skins/Vanjaro/Base.ascx.cs]: I'm not sure if we need the File.Exists check. I believe that's done internally in ClientResourceManager. Please confirm by removing the JS file and registering it with ClientResourceManager; there should not be any 404 in console.

**Review by meetmandeep — CHANGES_REQUESTED** [review]

File: Portals/_default/Skins/Vanjaro/Base.ascx.cs (hunk 1 above)
Comment (swanandpachode94) [reply Portals/_default/Skins/Vanjaro/Base.ascx.cs]: image
I can confirm it is throwing 404 error when debug mode is enabled.

Comment (meetmandeep) [reply Portals/_default/Skins/Vanjaro/Base.ascx.cs]: @swanand93  Okay. Let's refactor the ThemeManager.CurrentTheme to include ThemeManager.CurrenTheme.HasScript() to return true or false by doing a File.Exists check. It should also cache the result so it's only checked once.

**Review by Gaganbird — APPROVED** [review]

File: Portals/_default/Skins/Vanjaro/Base.ascx.cs [hunk 2]
```diff
@@ -125,7 +125,8 @@ protected void Page_Load(object sender, EventArgs e)
             JavaScript.RequestRegistration(CommonJs.jQuery, null, SpecificVersion.Latest);
 
             string ThemeJS = "~/Portals/_default/vThemes/" + Core.Managers.ThemeManager.CurrentTheme.Name + "/theme.js";
-            if (File.Exists(System.Web.Hosting.HostingEnvironment.MapPath(ThemeJS)))
+            
+            if (Core.Managers.ThemeManager.HasScript(ThemeJS))
```
Comment (meetmandeep) [inline Portals/_default/Skins/Vanjaro/Base.ascx.cs]: Pls refactor. There's no need to pass ThemeJS; ThemeManager can generate that based on the CurrentTheme.Name

**Review by meetmandeep — CHANGES_REQUESTED** [review]

**Review by meetmandeep — CHANGES_REQUESTED** [review]
See @Gaganbird for changes

File: Portals/_default/Skins/Vanjaro/Base.ascx.cs [hunk 3]
```diff
@@ -172,6 +154,37 @@ protected void Page_Load(object sender, EventArgs e)
             ResetModulePanes();
         }
 
+        private void InjectThemeJS()
+        {
+            if (Core.Managers.ThemeManager.HasThemeJS())
```
Comment (meetmandeep) [inline Portals/_default/Skins/Vanjaro/Base.ascx.cs]: Core.Managers.ThemeManager.HasThemeJS() should've been Core.Managers.ThemeManager.Current.HasThemeJS()

**Review by meetmandeep — CHANGES_REQUESTED** [review]

**Review by meetmandeep — APPROVED** [review]

**Pull request merged**